corehq/apps/es/domains.py

```diff
@@ -56,8 +62,48 @@ def mapping(self):
         return get_adapter_mapping(self)
 
     @classmethod
-    def from_python(cls, doc):
-        return from_dict_with_possible_id(doc)
+    def from_python(cls, domain_obj):
+        """
+        :param case: an instance of ``Domain``
+        :raises ``UnknownDocException`` if ``domain`` is not an instance of ``Domain``
+        """
+        from corehq.apps.domain.models import Domain
+        if not isinstance(domain_obj, Domain):
+            raise UnknownDocException(Domain, domain_obj)
+        return cls.from_multi(domain_obj)
+
+    @classmethod
+    def from_multi(cls, domain):
+        """
+        Takes in a ``Domain`` object or a domain dict
+        and applies required transformation to make it suitable for ES.
+        The function is replica of ``transform_domain_for_elasticsearch`` with added support for user objects.
+        In future all references to  ``transform_domain_for_elasticsearch`` will be replaced by `from_python`
+
+        :param user: an instance of ``Domain`` or ``dict`` which is ``domain.to_json()``
+
+        :raises UnknownDocException: if object passes in not instance of ``Domain``
+        """
+        from corehq.apps.accounting.models import Subscription
+        from corehq.apps.domain.models import Domain
+
+        def _verify_and_return_domain_as_dict():
+            if isinstance(domain, dict):
+                return deepcopy(domain)
```

Comment:
Suggestion: use `domain.copy()` (shallow copy) instead of `deepcopy(domain)` if possible.

This would require adjusting some code below such as
```py
doc_ret['deployment']['countries'] = []
```